### ISSUE TITLE
Not finding correct elbow

### ISSUE THREAD

**Issue by MattConflitti** [arvkevi/kneed#31]
Not entirely sure what the issue is on this one. The dataset has a very clear elbow. It is a decreasing, convex curve.

![](https://i.imgur.com/QwQy75G.png)

The calculated elbow is way off, but I can get the desired index by just grabbing the index of the maximum distance. Will this not always be the case?

**Comment by arvkevi** [comment]
Hi @MattConflitti thanks for posting, would you mind sharing which version you are using?
```python
import kneed
print(kneed.__version__)
```

**Comment by MattConflitti** [comment]
It looks like that attribute wasn't set.
```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-63-b9a19a10a516> in <module>()
      1 import kneed
----> 2 print(kneed.__version__)

AttributeError: module 'kneed' has no attribute '__version__'
```

I installed with pip and that says I have `kneed==0.3.1`

**Comment by arvkevi** [comment]
Great, that's the latest version. The last question, have you tried setting `interp_method='polynomial'` in `KneeLocator`?

**Comment by MattConflitti** [comment]
I did, this is the result. Similar to the previous one.
![](https://i.imgur.com/PwtshxG.png)

**Comment by arvkevi** [comment]
Thanks for checking.
Do you want to post `x, y` here so we're working off the same arrays?

You are correct, the knee should be located at the `max` of the distance curve `yd`.
Feel free to hack away on this bug with a pull request, otherwise, I'll try to reproduce the errors as soon as a can.

**Comment by MattConflitti** [comment]
The array is about 6000+ elements. If I have some time, I will do that. Thanks for the confirmation!

**Comment by arvkevi** [comment]
It should be easy enough to reproduce, thanks again, hope to jump on this soon.

**Comment by arvkevi** [comment]
This `x` and `y` reproduce a similar error:
```python
n = 6000
x = range(1, n+1)
y = sorted(np.random.gamma(0.5, 1.0, n), reverse=True)
```

**Comment by arvkevi** [comment]
This sparked a long-overdue refactor, working on this in #34

**Comment by arvkevi** [comment]
Hi, @MattConflitti thanks again for raising this issue, the package really needed this update! I spent some time re-reading the paper to ensure that I understood the algorithm correctly. The issue was that `kneed` was finding all knees and reporting the most recent (or last) knee point. As a result, setting the `S` parameter had no effect because the last knee was always returned.

I added your example to the README in #34 because it highlights how tuning the `S` parameter has an effect on the detected knee.
